### PR TITLE
News: new press release about GitHub Actions

### PR DESCRIPTION
**Pull request by neteler**
- text might still be improved, suggestions/edits welcome!

### CODE REVIEW

**Comment by neteler** [comment]
As usual, also a screenshot might be nice... ideas?

File: content/news/2020_06_28_github_actions.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+title: "New GRASS GIS source code analysis with GitHub Actions"
+date: 2020-06-28T12:12:12+01:00
+layout: "news"
+author: GRASS Development Team
+---
+
+### New GRASS GIS source code analysis with GitHub Actions
+
+After the successful migration from SVN to git earlier this year a new feature has be implemented and enabled: [GRASS GIS source code analysis with GitHub Actions](https://github.com/OSGeo/grass/actions) to build (compile), test, and perform Python static code analysis.
```
Comment (veroandreo) [inline content/news/2020_06_28_github_actions.md:10]:
```suggestion
After the successful migration from SVN to git earlier this year, a new feature has been implemented and enabled: [GRASS GIS source code analysis with GitHub Actions](https://github.com/OSGeo/grass/actions) to build (compile), test, and perform Python static code analysis.
```

File: content/news/2020_06_28_github_actions.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+The C and C++ source code is checked with GCC C/C++ standards check as well as additionally with gnu99 and c++11 tests. Python code is tested by a static code analysis/code quality check/linting using Flake8. The workflow builds and tests GRASS GIS on Ubuntu 16.04, 18.04 and Centos 7. An additional OSGeo4W workflow for GitHub Actions is currently in preparation. Success, errors and warnings are reported accordingly in the GitHub Actions interface for each pull request.
```
Comment (veroandreo) [inline content/news/2020_06_28_github_actions.md:12]:
```suggestion
The C and C++ source code is checked with GCC C/C++ standard checks and additionally with gnu99 and c++11 tests. Python code, onthe other hand, is tested with a static code analysis/code quality check/linting using Flake8. The workflow builds and tests GRASS GIS on Ubuntu 16.04, 18.04 and Centos 7. An additional OSGeo4W workflow for GitHub Actions is currently in preparation. Success, errors and warnings are reported accordingly in the GitHub Actions interface for each pull request.
```

File: content/news/2020_06_28_github_actions.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+date: 2020-06-28T12:12:12+01:00
+layout: "news"
+author: GRASS Development Team
+---
+
+### New GRASS GIS source code analysis with GitHub Actions
+
+After the successful migration from SVN to git earlier this year, a new feature has been implemented and enabled: [GRASS GIS source code analysis with GitHub Actions](https://github.com/OSGeo/grass/actions) to build (compile), test, and perform Python static code analysis.
+
+The C and C++ source code is checked with GCC C/C++ standard checks and additionally with gnu99 and c++11 tests. Python code, onthe other hand, is tested with a static code analysis/code quality check/linting using Flake8. The workflow builds and tests GRASS GIS on Ubuntu 16.04, 18.04 and Centos 7. An additional OSGeo4W workflow for GitHub Actions is currently in preparation. Success, errors and warnings are reported accordingly in the GitHub Actions interface for each pull request.
```
Comment (neteler) [inline content/news/2020_06_28_github_actions.md:12]:
```suggestion
The C and C++ source code is checked with GCC C/C++ standard checks and additionally with gnu99 and c++11 tests. Python code, on the other hand, is tested with a static code analysis/code quality check/linting using Flake8. The workflow builds and tests GRASS GIS on Ubuntu 16.04, 18.04 and Centos 7. An additional OSGeo4W workflow for GitHub Actions is currently in preparation. Success, errors and warnings are reported accordingly in the GitHub Actions interface for each pull request.
```

File: content/news/2020_06_28_github_actions.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
+---
+title: "New GRASS GIS source code analysis with GitHub Actions"
+date: 2020-06-28T12:12:12+01:00
+layout: "news"
+author: GRASS Development Team
+---
+
+### New GRASS GIS source code analysis with GitHub Actions
+
+After the successful migration from SVN to git earlier this year, a new feature has been implemented and enabled: [GRASS GIS source code analysis with GitHub Actions](https://github.com/OSGeo/grass/actions) to build (compile), test, and perform Python static code analysis.
```
Comment (mlennert) [inline content/news/2020_06_28_github_actions.md:10]:
```suggestion
After the successful migration from SVN to git earlier this year, a new feature has been implemented and enabled: [GRASS GIS source code analysis with GitHub Actions](https://github.com/OSGeo/grass/actions) to build (compile), test, and perform Python static code analysis, in order to improve the overall quality, maintainability and standard compliance of our code.
```

**Comment by neteler** [comment]
Yes, we want to inform our user base - so more introductory explanations are welcome.

File: content/news/2020_06_28_github_actions.md [hunk 5]
```diff
@@ -0,0 +1,16 @@
+---
+title: "New GRASS GIS source code analysis with GitHub Actions"
+date: 2020-06-28T12:12:12+01:00
+layout: "news"
+author: GRASS Development Team
+---
+
+### New GRASS GIS source code analysis with GitHub Actions
+
+After the successful migration from SVN to git earlier this year, a new feature has been implemented and enabled: [GRASS GIS source code analysis with GitHub Actions](https://github.com/OSGeo/grass/actions) to build (compile), test, and perform Python static code analysis, in order to improve the overall quality, maintainability and standard compliance of our code.
```
Comment (neteler) [inline content/news/2020_06_28_github_actions.md:10]:
```suggestion
After the successful migration from SVN to git earlier this year ([here](/news/2019_05_17_svn_to_github_migration_started/) and [here](/news/2020_01_23_grass_gis_issues_now_on_github/)), a new feature has been implemented and enabled: [GRASS GIS source code analysis with GitHub Actions](https://github.com/OSGeo/grass/actions) to build (compile), test, and perform Python static code analysis, in order to improve the overall quality, maintainability and standard compliance of our code.
```

**Comment by neteler** [comment]
@HuidaeCho could you please review the part about GitHub Actions and OSGeo4W?  (of course other comments/additions are welcome as well)
thanks

File: content/news/2020_06_28_github_actions.md [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+date: 2020-06-28T12:12:12+01:00
+layout: "news"
+author: GRASS Development Team
+---
+
+### New GRASS GIS source code analysis with GitHub Actions
+
+After the successful migration from SVN to git earlier this year ([here](/news/2019_05_17_svn_to_github_migration_started/) and [here](/news/2020_01_23_grass_gis_issues_now_on_github/)), a new feature has been implemented and enabled: [GRASS GIS source code analysis with GitHub Actions](https://github.com/OSGeo/grass/actions) to build (compile), test, and perform Python static code analysis, in order to improve the overall quality, maintainability and standard compliance of our code.
+
+The C and C++ source code is checked with GCC C/C++ standard checks and additionally with gnu99 and c++11 tests. Python code, on the other hand, is tested with a static code analysis/code quality check/linting using Flake8. The workflow builds and tests GRASS GIS on Ubuntu 16.04, 18.04 and Centos 7. An additional OSGeo4W workflow for GitHub Actions is currently in preparation. Success, errors and warnings are reported accordingly in the GitHub Actions interface for each pull request.
```
Comment (HuidaeCho) [inline content/news/2020_06_28_github_actions.md:12]: The OSGeo4W workflow is already committed and in place. I'm planning to copy out the built package as [a GitHub artifact](https://help.github.com/en/actions/configuring-and-managing-workflows/persisting-workflow-data-using-artifacts) for daily builds, if needed, as we discussed in [#692](https://github.com/OSGeo/grass/pull/692). Having said that, I'm not sure if that's really necessary when we already have a working build process for daily OSGeo4W builds (where?).

Comment (hellik) [reply content/news/2020_06_28_github_actions.md:12]: >I'm not sure if that's really necessary when we already have a working build process for daily OSGeo4W builds (where?).  

daily OSGeo4W builds are done at @landam server. 

in the long run, I suggest to use such things like GH action/GH arctifacts (or something else) to ease the burden from 1 maintainer of daily builds to a group.

AFAIK there are ideas to build/setup OSGeo4W itself in something similar like azure etc. thus, if GRASS is ready, the better.

@landam what do you think?

Comment (neteler) [inline content/news/2020_06_28_github_actions.md:12]:
```suggestion
The C and C++ source code is checked with GCC C/C++ standard checks and additionally with gnu99 and c++11 tests. Python code, on the other hand, is tested with a static code analysis/code quality check/linting using Flake8. The workflow builds and tests GRASS GIS on Ubuntu 16.04, 18.04, Centos 7 as well as OSGeo4W. Success, errors and warnings are reported accordingly in the GitHub Actions interface for each pull request.
```

Comment (neteler) [reply content/news/2020_06_28_github_actions.md:12]: ... like this?

Comment (HuidaeCho) [reply content/news/2020_06_28_github_actions.md:12]: If that's the case, +1 for daily artifacts.

Comment (HuidaeCho) [reply content/news/2020_06_28_github_actions.md:12]: Looks good!

**Comment by neteler** [comment]
Anyone willing to add some intro sentences? Otherwise I'll merge...

File: content/news/2020_06_28_github_actions.md (hunk 6 above)
Comment (wenzeslaus) [reply content/news/2020_06_28_github_actions.md:12]: > The C and C++ source code is checked with GCC C/C++ standard checks and additionally with gnu99 and c++11 tests. 

Notably, we don't check yet ISO or strict compliance, it is so far just a rough check, so saying something like *GCC C/C++ checks including GNU C11 and C++11* is probably better.

> Python code, on the other hand, is tested with a static code analysis/code quality check/linting using Flake8.

...with tens of messages disabled. Can you please add a note about seeking contributors to reduce the number of remaining issues?

Comment (neteler) [reply content/news/2020_06_28_github_actions.md:12]: Done so, thanks for the suggestions

File: content/news/2020_06_28_github_actions.md [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+The C and C++ source code is checked with GCC C/C++ checks including GNU C11 and C++11. For Python code, an initial static code analysis/code quality check/linting using Flake8 has been implemented. The workflow builds and tests GRASS GIS on Ubuntu 16.04, 18.04, Centos 7 as well as OSGeo4W. Success, errors and warnings are reported accordingly in the GitHub Actions interface for each pull request.
+
+We seek contributors for extending our tests! Please don't hesitate to submit comments or code editions [here](https://github.com/OSGeo/grass/tree/master/.github/workflows). 
+
+This code analysis has been set up as a continuous integration (CI) workflow, i.e. after each source code change - when submitting a "pull request" - it is triggered automatically.
+
+Besides core GRASS GIS ([master](https://github.com/OSGeo/grass/) and [release_branch_7_8](https://github.com/OSGeo/grass/tree/releasebranch_7_8)) also the [addons repository](https://github.com/OSGeo/grass-addons/actions) has a minimalistic set of source code analysis enabled.
```
Comment (neteler) [inline content/news/2020_06_28_github_actions.md:18]:
```suggestion
This code analysis has been set up as a continuous integration (CI) workflow, i.e. after each source code change - when submitting a "pull request" - it is triggered automatically.

Besides core GRASS GIS ([master](https://github.com/OSGeo/grass/) and [release_branch_7_8](https://github.com/OSGeo/grass/tree/releasebranch_7_8)) also the [addons repository](https://github.com/OSGeo/grass-addons/actions) has a minimalistic set of source code analysis enabled.

We seek contributors for extending our tests! Please don't hesitate to submit comments or code editions [here](https://github.com/OSGeo/grass/tree/master/.github/workflows). 
```

File: content/news/2020_06_28_github_actions.md (hunk 6 above)
Comment (neteler) [reply content/news/2020_06_28_github_actions.md]: Updated accordingly

**Pull request merged**